### DISCUSSION TITLE
System theme

### DISCUSSION THREAD

**Discussion by alternateved** [material-shell/material-shell#618] [Ideas]
Hello all and also @PapyElGringo, 

First of all, thank you very much for your hard work. Your material-shell is amazing. It takes GNOME experience to a whole new level. 

Secondly, I have a small question or suggestion - would it be possible for you to implement option so that shell would know what theme is currently switched on in the system? This type of functionality is enabled in modern browsers and I also spotted it in gnome-terminal so that whenever theme is switched for the light or dark one, browser and terminal switch its theme accordingly. 

I am asking this question because I would love to use your extension with another one called "Night Theme Switcher" which changes the theme accordingly to the time of the day. 

Thanks once again for your creation!

**Comment by PapyElGringo** [comment]
It's would be interesting to investigate more on how browser and such app are getting theming values from the system because right know with the Gnome Shell theme I don't see how I could do that.

**Reply by alternateved** [reply]
Hmm, I found some information. When it comes to browser, there is a combination of JavaScript and CSS. Code below could be helpful in that regard:

`if (window.matchMedia && !!window.matchMedia('(prefers-color-scheme: dark)').matches) {
    console.info('Dark Theme detected')
} else {
  console.info('Light Theme detected')
}`

In Gnome Shell, current theme could be recognized by below terminal command:

`gsettings get org.gnome.desktop.interface gtk-theme`

So there could be some conditional which detects if the name of current theme contains "dark" word. I hope it does help anyhow.

**Reply by PapyElGringo** [reply]
We don't only want to know if it's a dark theme or a light theme we would like get the main colors of the theme. But the main difference between the browser and us it's that the browser is reading the GTK theme and we are supposed to read the Shell theme. But the Shell theme is a css theme that patch the colors of specific named item like the panel and such. Since we use our own elements the legacy css themes can't theme our components.

We may be able to follow the gtk theme instead of the Shell theme but I don't know how we could read it easily
